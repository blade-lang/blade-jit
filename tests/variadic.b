echo '------------ Test 1 ------------'
def test(...a) {
  echo a
}

test(1, 2, 3)

echo '------------ Test 2 ------------'
def test2(a, b, ...c) {
  echo a
  echo b
  echo c
}

test2(10, 20, 'hello', [1,2,3])

echo '------------ Test 3 ------------'
def test3(a, b, c, ...d) {
  echo a
  echo b
  echo c
  echo d
}

test3(10)
echo '------------ Test 4 ------------'
test3(10, 12)
echo '------------ Test 5 ------------'
test3(10, 12, 14)
