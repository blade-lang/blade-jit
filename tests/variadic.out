------------ Test 1 ------------
[1, 2, 3]
------------ Test 2 ------------
10
20
[hello, [1, 2, 3]]
------------ Test 3 ------------
10
nil
nil
[]
------------ Test 4 ------------
10
12
nil
[]
------------ Test 5 ------------
10
12
14
[]